intf/accessor_linux.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intf
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/vishvananda/netlink"
+)
+
+func init() {
+	// On init when the package has been built on Linux, load the netlink
+	// accessor as the implementation to be used.
+	//
+	// TODO(robjs): Consider whether we should use build tags vs. solely
+	// the underlying system, since this would allow a build tag to say
+	// that the package should use gRPC wire.
+	accessor = netlinkAccessor{}
+}
+
+// netlinkAccessor implements the NetworkAccessor interface for a Linux
+// system.
+type netlinkAccessor struct {
+	unimplementedAccessor
```

Comment:
i don't think you should embed the "fake" into the real thing
just provide the direct methods only

Reply:
The intention here was to ensure that we implement the interface. We can mandate that all accessors implement the interface with methods themselves, but this approach looks a little "neater" to me.

I guess if we want to have a subset of methods implemented generally, then we should do this through defining a smaller interface. This complicates things a bit for library users though.